com.unity.netcode.gameobjects/Components/HalfVector3.cs

```diff
@@ -0,0 +1,114 @@
+using System.Runtime.CompilerServices;
+using UnityEngine;
+
+namespace Unity.Netcode.Components
+{
+    /// <summary>
+    /// Half float precision <see cref="Vector3"/>
```

Comment:
Based on our previous conversation, could you make a comment here about how the ushorts in this class are binary reinterpretations of half-floats because C# doesn't support them at the language level? Also in the other cases where you use this technique. Would be good to know for anyone who comes and looks at it later.